backend/src/server_host.py

```diff
@@ -152,6 +157,102 @@ async def get_features(request: Request):
     return await worker.proxy_request(request)
 
 
+def deps_to_dep_info(deps: list[api.Dependency]) -> list[DependencyInfo]:
+    return [
+        DependencyInfo(
+            package_name=dep.pypi_name,
+            display_name=dep.display_name,
+            version=dep.version,
+            extra_index_url=dep.extra_index_url,
+        )
+        for dep in deps
+    ]
+
+
+@app.route("/packages/uninstall", methods=["POST"])
+async def uninstall_dependencies_request(request: Request):
+    full_data = dict(request.json)  # type: ignore
+    package_to_uninstall = full_data["package"]
+
+    packages = await worker.get_packages()
+
+    package = next((x for x in packages if x.name == package_to_uninstall), None)
+
+    if package is None:
+        return json(
+            {"status": "error", "message": f"Package {package_to_uninstall} not found"},
+            status=404,
+        )
+
+    def update_progress(
+        message: str, progress: float, status_progress: float | None = None
+    ):
+        return AppContext.get(request.app).setup_queue.put_and_wait(
```

Comment:
Any reason why you wait here? What does it matter to the backend whether the frontend read the status or not?

Reply:
Nope. I just copy and pasted another line. I probably made this same mistake in other PRs as well actually

Reply:
Ok so for some reason it only works like this and not at all with put, and i can't explain why